packages/astro/src/types/public/integrations.ts

```diff
@@ -245,13 +243,52 @@ export interface AstroIntegration {
 
 /**
  * A smaller version of the {@link RouteData} that is used in the integrations.
+ * @deprecated Use {@link IntegrationResolvedRoute}
  */
 export type IntegrationRouteData = Omit<
 	RouteData,
-	'isIndex' | 'fallbackRoutes' | 'redirectRoute'
+	'isIndex' | 'fallbackRoutes' | 'redirectRoute' | 'origin'
 > & {
 	/**
 	 * {@link RouteData.redirectRoute}
 	 */
 	redirectRoute?: IntegrationRouteData;
 };
+
+export interface IntegrationResolvedRoute
+	extends Omit<
+		RouteData,
+		| 'isIndex'
+		| 'fallbackRoutes'
+		| 'redirectRoute'
+		| 'route'
+		| 'pattern'
+		| 'component'
+		| 'prerender'
+		| 'distURL'
+	> {
```

Comment:
(optional) Maybe this could use `Pick` so we don't accidentally expose new props